src/queries/label-text.js

```diff
@@ -12,30 +12,73 @@ import {
 } from './all-utils'
 import {queryAllByText} from './text'
 
+function getCombinations(labels, matcher) {
+  const combs = [[]]
+  const matching = []
+  for (const label of labels) {
+    const copy = [...combs] // See note below.
```

Comment:
I'm not sure what note this is referring to. Also, if I'm following this logic correctly, the for loop on the next line will only be run only once (for this iteration of the nested loop) and the value of the `prefix` will be `[]` every time. Any reason we need a loop here?

Reply:
I am not sure I am totally getting your point, are you telling that [this](https://github.com/testing-library/dom-testing-library/blob/732a8abf5560b13cfba976f5f801048e7018e2d4/src/queries/label-text.js#L20) loop runs only once inside the [outer](https://github.com/testing-library/dom-testing-library/blob/732a8abf5560b13cfba976f5f801048e7018e2d4/src/queries/label-text.js#L18) loop?

Reply:
Correct. Because `copy` is `[[]]` and so it's an array with only one element in it. It'll never be....

Wait. I guess that's only the case on the first iteration of the outer loop. Future iterations could have more... Hmmm... I need to take more time to understand what's going on here...

Reply:
Yes, I think it's only the first iteration of the outer loop that makes the inner one run only once.